### DISCUSSION TITLE
Printer Cfg for Creality Sonic Pad

### DISCUSSION THREAD

**Discussion by mackmind** [1coderookie/Klipper4KobraGoNeo#7] [Announcements]
Hello,
I've just managed to set up Kobra Neo with Creality Sonic Pad and I had to do some tweaks to make it run that involved basically merging existing configs shared here (thanks!) with default file from Creality and with themselves, so:
- in [stepper_z] endstop_pin: probe:z_virtual_endstop - since SonicPad didn't read Z endstop from Neo Z endstop sensor and either went beyond range or smashed nozzle into the bed. Fortunately LeviQ probe works well as Z endstop.
- no [include] files, this caused error on initialisation,
- safe_z_home at home_xy_position: 110,110 in middle of the plate - so that in case SonicPad smashes nozzle into printbed, then at least it's not in the corner. Also thanks to this printbed level value mid-bed should be closest to 0.
- in [extruder] max_extrude_cross_section: 50 so there's no 'safety'  Klipper extrusion error that blocks prints from continuing
- extruder rotation distance slightly altered (+1%) per my rough e-step calibration result
- homing and probing speeds increased, while tolerance lowered
- other speeds increased, so that they can be freely adjusted later in UI instead of being hardcoded in cfg.

[NEO-mackmind-SonicPad-printer.cfg.txt](https://github.com/1coderookie/Klipper4KobraGoNeo/files/11481301/NEO-mackmind-SonicPad-printer.cfg.txt)

Hope this helps someone - I'm a noob in FDM and I did changes despite that, so this CFG may need corrections. The CFG is what I select when adding a new printer - then SonicPad's Klipper does it's modifications, so the CFG that is later seen through WebUI is slightly different (and clearly different after printer calibration).

With this installation of SonicPad for Kobra Neo should be as follows:
1. boot Kobra with Klipper.bin  (as firmware.bin) from this repository and instructions (thanks)
In my case there was no progress message on any screen - printer has to be left for 10-15 min to be sure that it installed Klipper even though the file is tiny, then restarted.
2. plug in Kobra Neo to SonicPad USB port1 - or change number in printer.cfg if you want to use another USB:
[mcu]
serial: /dev/serial/by-id/usb-serial_1
3. Initialise per SonicPad instruction for non-standard printers with printer connected (and turned on) and USB with printer.cfg in main USB stick directory (root) plugged in.
4. Follow as usual per SonicPad instructions.

**Comment by 1coderookie** [comment]
Great! Thanks a lot!  
I'll add your cfg to the repo and link to this discussion from the belonging chapter of the infopage after adding the SonicPad.  

> no [include] files, this caused error on initialisation,  

FYI: this *might* have been caused by a missing file tho. I have the include command active in the cfg for a file which I didn't upload. So it might even work with those includes, if the belonging files really are given. However, I'll change that in my cfg (or add the belonging files to the repo as well).
